macOS/DuckDuckGo/Favicons/Model/FaviconReferenceCache.swift

```diff
@@ -255,26 +223,27 @@ final class FaviconReferenceCache: FaviconReferenceCaching {
                      tld: TLD) async {
         // Remove host references
         let bookmarkedHosts = bookmarkManager.allHosts()
-        await removeHostReferences(filter: { hostReference in
+        await removeHostReferences { hostReference in
             let host = hostReference.host
             let baseDomain = tld.eTLDplus1(host) ?? ""
             return baseDomains.contains(baseDomain) && !bookmarkedHosts.contains(host) && !logins.contains(host) && !history.contains(host)
-        }).value
+        }
         // Remove URL references
-        await removeUrlReferences(filter: { urlReference in
+        await removeUrlReferences { urlReference in
             guard let host = urlReference.documentUrl.host else {
                 return false
             }
             return baseDomains.contains(host) && !bookmarkedHosts.contains(host) && !logins.contains(host) && !history.contains(host)
-        }).value
+        }
     }
 
     // MARK: - Private
 
+    @MainActor
     private func insertToHostCache(faviconUrls: (smallFaviconUrl: URL?, mediumFaviconUrl: URL?), host: String, documentUrl: URL) {
```

Comment:
For these functions that spawn tasks I assumed we do it to improve performance as these functions don’t need to wait for updating the store, so I kept the tasks as is (adding `detached` everywhere because only some of them used Task.detached). Otherwise we could make functions async and await these tasks.